src/components/cylc/cylcObject/Menu.vue

```diff
@@ -166,43 +175,26 @@ export default {
   },
 
   computed: {
+    primaryMutations () {
+      return this.$workflowService.primaryMutations[this.type] || []
+    },
     canExpand () {
-      if (!this.mutations) {
-        return false
-      }
-      return !!this.$workflowService.primaryMutations[this.type]
+      return this.primaryMutations.length && this.mutations.length > this.primaryMutations.length
     },
     ...mapState('user', ['user']),
-    userPermissions () {
-      return this.user.permissions.map((str) => str.toLowerCase())
-    },
-    authorizedMutations () {
-      return this.mutations
-        .map(mutation => {
-          if (this.userPermissions.includes(mutation[0].name.toLowerCase())) {
-            mutation[2] = true
-          }
-          return mutation
-        })
-    },
     displayMutations () {
-      if (!this.mutations || this.user.permissions.length < 2) {
+      if (!this.mutations.length || this.user.permissions.length < 2) {
```

Comment:
Remember that empty arrays are truthy in JavaScript! Always use `.length` for testing conventional truthiness otherwise you're just testing that it is not `undefined`